code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -80,18 +80,24 @@ This used to be handled by attackby() on the light fixtures and bulbs themselves
 	if(!gather_loc || !isturf(gather_loc))
 		return 0
 	var/obj/machinery/light/lightfixture = locate() in gather_loc.contents
-	var/obj/item/weapon/light/best_light = get_best_light(lightfixture)
-	if(lightfixture && lightfixture.current_bulb && is_light_better(best_light, lightfixture.current_bulb))
-		. = ReplaceLight(lightfixture, usr)
-		return .
-	else if(lightfixture && !lightfixture.current_bulb)
-		. = ReplaceLight(lightfixture, usr)
-		if(.)
-			return .
-	else
+	if(!lightfixture)
+		// No light fixture found, try to gather light items from the turf
 		for(var/obj/O in gather_loc.contents)
 			. = insert_if_possible(O)
+			if(.)
+				return .
```

Comment:
Your change makes it so that clicking on a turf that contains multiple lights no longer sucks up every light on the turf, but just one at a time.